rclcpp/src/rclcpp/executor.cpp

```diff
@@ -266,7 +266,7 @@ Executor::execute_subscription(
   if (ret == RCL_RET_OK) {
     message_info.from_intra_process = false;
     subscription->handle_message(message, message_info);
-  } else if (ret != RCL_RET_SUBSCRIPTION_TAKE_FAILED) {
+  } else if (ret == RCL_RET_SUBSCRIPTION_TAKE_FAILED) {
```

Comment:
I think the previous comparison is correct and shouldn't be changed.

`rcl_take` returns `RCL_RET_SUBSCRIPTION_TAKE_FAILED` if there was no error but no message to take (see https://github.com/ros2/rcl/blob/88db558aa303ad29f24cc47da28f27f64a27eaa4/rcl/src/rcl/subscription.c#L146). In that case there should be no error message. Only in all other cases there should be a message printed to `stderr`. The constant would probably be more appropriately names `RCL_RET_SUBSCRIPTION_NOTHING_TAKEN` (or similar).